### PR TITLE
Add meter method which returns remaining computation for given computation kind

### PR DESCRIPTION
**Pull request by turbolent**
For https://github.com/onflow/cadence/issues/2853. See https://github.com/onflow/cadence/issues/2853#issuecomment-1937112167

### CODE REVIEW

**Review by ramtinms — APPROVED** [review]

File: fvm/meter/computation_meter.go [hunk 1]
```diff
@@ -116,10 +116,24 @@ func (m *ComputationMeter) ComputationAvailable(
...
+// ComputationRemaining returns the remaining computation left in the transaction for the given type
+func (m *ComputationMeter) ComputationRemaining(kind common.ComputationKind) uint {
+	w, ok := m.params.computationWeights[kind]
+	// if not found return has capacity
+	// given the behaviour of MeterComputation is ignoring intensities without a set weight
+	if !ok {
+		return math.MaxUint
+	}
+
+	remainingComputationUsage := m.params.computationLimit - m.computationUsed
```
Comment (janezpodhostnik) [inline fvm/meter/computation_meter.go:133]: I think we should not rely on computation limit to be `>=` to computation used here.

**Comment by bluesign** [comment]
Is this stale?

**Comment by janezpodhostnik** [comment]
@bluesign ComputationAvailable was added in another PR https://github.com/onflow/flow-go/pull/4861/files#diff-f0abf60d6957bfe1c0f9a9c50108652f85d6515ca786892300f48b76e0144f86R100

File: fvm/meter/computation_meter.go (hunk 1 above)
Comment (turbolent) [reply fvm/meter/computation_meter.go:133]: Good point! Added handling for this edge case in f075976943eb85ee0b111e4c89285bb45367af99

**Comment by turbolent** [comment]
@onflow/flow-cadence-execution PTAL

**Review by ramtinms — APPROVED** [review]

**Pull request merged**